servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingHttpClientToHttpClient.java

```diff
@@ -44,26 +44,21 @@ public HttpExecutionStrategy executionStrategy() {
 
     @Override
     public Single<HttpResponse> request(final HttpRequest request) {
-        return Single.defer(() -> request(requestStrategy(request, strategy), request).subscribeShareContext());
+        return Single.defer(() -> {
+            request.context().putIfAbsent(HTTP_EXECUTION_STRATEGY_KEY, strategy);
+            return client.request(request.toStreamingRequest())
+                    .flatMap(response -> response.toResponse().subscribeShareContext());
```

Comment:
seems like the `subscribeShareContext` be applied to the outside of flatMap `.flatMap(response -> response.toResponse()).subscribeShareContext();`?

Reply:
👀 
Based on my reply https://github.com/apple/servicetalk/pull/1960#discussion_r752746456 I believe we need `subscribeShareContext()` in both places (inside `flatMap` and inside `defer`).